x-pack/platform/plugins/shared/streams/server/routes/streams/crud/validate_classic_stream_route.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from '@kbn/zod';
+import { isNotFoundError } from '@kbn/es-errors';
+
+import { STREAMS_API_PRIVILEGES } from '../../../../common/constants';
+import { createServerRoute } from '../../create_server_route';
+import { findConflictingTemplates } from '../../../lib/streams/helpers/validate_template_conflicts';
+
+export const validateClassicStreamRoute = createServerRoute({
+  endpoint: 'POST /internal/streams/_validate_classic_stream',
+  options: {
+    access: 'internal',
+    summary: 'Validate classic stream name',
+    description:
+      'Validates whether a classic stream name can be used by checking for duplicate streams, existing data streams, and higher priority index template conflicts',
+    availability: {
+      stability: 'experimental',
+    },
+  },
+  security: {
+    authz: {
+      requiredPrivileges: [STREAMS_API_PRIVILEGES.manage],
+    },
+  },
+  params: z.object({
+    body: z.object({
+      name: z.string(),
+      selectedTemplateName: z.string(),
+    }),
+  }),
+  handler: async ({ params, request, getScopedClients, logger }) => {
+    const { streamsClient, scopedClusterClient } = await getScopedClients({ request });
+    const { name, selectedTemplateName } = params.body;
+
+    // Check 1: Does a classic stream with this name already exist?
+    const streamExists = await streamsClient.existsStream(name);
+    if (streamExists) {
+      return {
+        isValid: false,
+        errorType: 'duplicate' as const,
+      };
+    }
+
+    // Check 2: Does a data stream with this name already exist?
+    try {
+      const dataStreamResult = await scopedClusterClient.asCurrentUser.indices.getDataStream({
```

Comment:
This shouldn't be necessary - `streamsClient.existsStream` should handle this case already